INVOKER_VALIDATION.md

```diff
@@ -0,0 +1,180 @@
+
+#### Sending a Request / Getting a Reply
+
+In another terminal, send a HTTP request against `localhost:8080` and wait for the response. For instance:
+
+```shell
+curl http://localhost:8080 -H'Content-Type:application/json' -d'0' -H'Accept:text/plain' -v
+```
+
+#### Publishing inputs / Observing outputs
+
+For that part, you will need [a working riff installation](https://projectriff.io/docs), **including the streaming runtime**.
+
+Once that's done, create the required gateways and streams for the function to test with the riff CLI.
+For instance:
+
+```shell
+ $ riff streaming kafka-gateway create franz --bootstrap-servers kafka.kafka:9092 --tail
+ $ riff streaming stream create words --content-type text/plain --gateway franz
+ $ riff streaming stream create numbers --content-type application/json --gateway franz
+ $ riff streaming stream create repeated --content-type text/plain --gateway franz
+```
+
+Now, clone the [Java Processor](https://github.com/projectriff/streaming-processor) and run its local bindings generator
+in order to locally replicate the binding structure generated on the cluster by the commands above:
+
+```shell
+ $ cd ~/workspace/streaming-processor
+# topic declaration order possibly matters if the targetted function resolves arguments by order.
+ $ ./src/etc/local_bindings.sh \
+    --default-gateway franz-gateway-4v8fj:6565 \
+    --input-topic words \
+    --input-topic numbers \
+    --output-topic repeated --accept text/plain
+Please configure CNB_BINDINGS envvar to point to /var/folders/t5/04x3n9gx2c1b74vrljr4sn6h0000gq/T/local-bindings5922978358290298314 when running Processor
+``` 
+
+Note the generated path, as we will reuse it when configuring the streaming processor.
+You can also point to an existing directory with `--base-directory`.
+
+> Just run `./src/etc/local_bindings.sh` to see the available configuration options.
+
+Install and run [`kubefwd`](https://kubefwd.com/) in **another** terminal so that `franz-gateway-4v8fj:6565` is 
+forwarded to the actual in-cluster Liiklus instance. In this example:
+
+```shell
+ $ sudo kubefwd svc franz-gateway-4v8fj
+```
+
+The invoker must have started, otherwise, please refer to previous sections.
+
+Once the invoker is started, start the processor **in a separate terminal** with the appropriate configuration. 
+In this example (notice the `CNB_BINDINGS` value from the `local_bindings.sh` execution):
+
+```shell
+ $ INPUT_NAMES='words,numbers' \
+    INPUT_START_OFFSETS='latest,latest' \
+    OUTPUT_NAMES='repeated' \
+    FUNCTION='localhost:8081' \
+    GROUP='somehow-irrelevant' \
+    CNB_BINDINGS='/var/folders/t5/04x3n9gx2c1b74vrljr4sn6h0000gq/T/local-bindings5922978358290298314' \
+    mvn exec:java -Dstart-class="io.projectriff.processor.Processor"
+```
+
+Finally, clone the [`stream-client-go` fork](https://github.com/fbiville/stream-client-go/tree/stream_publisher):
```

Comment:
I'm not sure we want to go ahead with this :)

Reply:
I suggest using dev-utils in cluster to publish:
https://projectriff.io/docs/v0.6/runtimes/streaming#testing-with-dev-utils